### DISCUSSION TITLE
Reviewers vs assignee on automergeable PR's (especially on Bitbucket Server)

### DISCUSSION THREAD

**Discussion by nickstolwijk** [renovatebot/renovate#9355] [Request Help] [answered]
It seems reviewers are always added to a PR, while assignees are only assigned to automergeable PR's when it fails status checks. Unfortunately, when on Bitbucket Server, assignees are unsupported. [1]

Is it possible to configure Renovate to only assign a reviewer/assignee to a PR if it fails status checks (on Bitbucket Server)?

If not, what would make more sense:
- adding assignees to a Bitbucket-server as if they were a reviewer (and use  the assignAutomerge flag to toggle the functionality)
- adding a flag to reviewers to only add them if the automergeable PR fails status checks (like assignAutomerge)
- something else entirely

[1] https://github.com/renovatebot/renovate/blob/master/lib/platform/bitbucket-server/index.md

**Comment by rarkins** [comment, marked as answer]
I'm surprised if reviewers are always added to PRs, including if they are automerging. Are you sure on this matter?

**Reply by nickstolwijk** [reply]
![image](https://user-images.githubusercontent.com/1562274/113397160-e456b100-939c-11eb-8f06-4e488ad09fae.png)

It seems that is what it is. :-(

renovate.json:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    "ex/example-renovate-configuration:automergeSDE"
  ],
  "reviewers": [ "stolwn1" ]
}
```
automergeSDE.json:
```
{
    "packageRules": [
        {
            "matchPackagePatterns": [ "^nl.example:example*" ],
            "groupName": "Example Parents",
            "automerge": true
        }
    ]
}
```
Nothing about assignees or reviewers in config.js.

**Reply by viceice** [reply]
Are you using https://docs.renovatebot.com/configuration-options/#bbusedefaultreviewers. They are always assigned on pr creation

**Reply by nickstolwijk** [reply]
```
module.exports = {
    platform: 'bitbucket-server',
    username: process.env.BITBUCKET_RENOVATE_CREDS_USR,
    password: process.env.BITBUCKET_RENOVATE_CREDS_PSW,
    endpoint: 'https://git.example.com/',
    baseBranches: ['develop'],
    rebaseWhen: 'behind-base-branch',
    onboarding: false,
    autodiscover: true,
    autodiscoverFilter: '+(bib|etc)/*',
    timezone: "Europe/Amsterdam",
    hostRules: [
        {
            baseUrl: 'https://nexus.example.com',
            username: process.env.NEXUS_RENOVATE_CREDS_USR,
            password: process.env.NEXUS_RENOVATE_CREDS_PSW,
        }
    ],
    packageRules: [
        {
            'matchDatasources': ['maven'],
            'registryUrls': ['https://example.com/repository/public/']
        },
        {
            'matchDatasources': ['npm'],
            'registryUrls': ['https://nexus.example.com/repository/npm-group/']
        }
    ]
};
```
As far as I can tell, we didn't enable that.

**Reply by rarkins** [reply]
Please capture the debug logs the next time a PR is created, if it's assigned. I am still very surprised if Renovate is directly responsible.

**Reply by nickstolwijk** [reply]
I took a better look and the offending repositories had default reviewers assigned from inside bitbucket. I cried wolf too soon. :-)

**Comment by bjbr-dev** [comment]
We seem to have encountered this as well. We'd like to reduce the number of notifications we're getting from renovate since we're starting to ignore pull requests from developers.

this is our renovate.json5 in the repo
```
{
    $schema: "https://docs.renovatebot.com/renovate-schema.json",
    extends: ["config:js-lib", "group:dotNetCore"],
    baseBranches: ["develop"],
    lockFileMaintenance: { enabled: true },

    prConcurrentLimit: 4,
    branchConcurrentLimit: 4,

    packageRules: [
        {
            groupName: "Mongo packages",
            matchPackagePatterns: ["^mongo"],
        },
        {
            matchPackageNames: ["jest"],
            enabled: false,
        },
        {
            matchUpdateTypes: ["minor", "patch", "pin", "digest"],
            automerge: true,
            reviewers: [],
        },
    ],
}
```

and this is the renovate config in the runner:
```
module.exports = {
  platform: "bitbucket",
  username: process.env.RENOVATE_BITBUCKET_USERNAME,
  onboardingConfig: {
    extends: ["config:base"],
  },
  repositories: ["redacted"],
};
```
